wire-ios-sync-engine/Source/Repositories/UserRepository_.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireDataModel
+
+/// Facilitate access to users related domain objects.
+///
+/// A repository provides an abstraction for the access and storage
+/// of domain models, concealing how and where the models are stored
+/// as well as the possible source(s) of the models.
+
+protocol UserRepositoryProtocol_ {
+
+    /// Fetch and persist all locally known users
+
+    func pullKnownUsers() async throws
+
+    /// Fetch and persist a list of users
+    ///
+    /// - parameters:
+    ///     - userIDs: IDs of users to fetch
+
+    func pullUsers(userIDs: [WireDataModel.QualifiedID]) async throws
+
+}
+
+final class UserRepository_: UserRepositoryProtocol_ {
```

Comment:
I am not sure why the underscores are required?

Reply:
Because there's was already a UserRepository in `WireSyncEngine` the two should be merged but I didn't want to do it in this PR.

But since the `WireDomain` module has now been merged I can probably drop the _. I'll try

Reply:
Alright, I merged the functionality from the old UserRepository into the new one, but it turns it's not so easy to adopt the new one in `SupportedProtocolsService` since the new one depends on `UsersAPI` and we currently don't have good way to create instances of APIs.

We need to know the API version to build api structs but we fetch it lazily. Until we have a solution for that I'll not try to bring in these new repositories into our old system.

Reply:
Thanks of trying, we can figure it out later.